test/e2e/e2e_test.go

```diff
@@ -99,7 +99,26 @@ var _ = ginkgo.Describe("leaderWorkerSet e2e tests", func() {
 		testing.ExpectLeaderWorkerSetAvailable(ctx, k8sClient, lws, "All replicas are ready")
 		testing.UpdateWorkerTemplate(ctx, k8sClient, lws)
 
-		testing.ExpectValidLeaderStatefulSet(ctx, lws, k8sClient)
+		testing.ExpectValidLeaderStatefulSet(ctx, k8sClient, lws, 2)
+		testing.ExpectValidWorkerStatefulSets(ctx, lws, k8sClient, true)
+		testing.ExpectValidPods(ctx, k8sClient, lws, &corev1.PodList{})
+		// Wait for leaderWorkerSet to be ready again.
+		testing.ExpectLeaderWorkerSetAvailable(ctx, k8sClient, lws, "All replicas are ready")
+	})
+
+	ginkgo.It("Can perform a rolling update with maxSurge set", func() {
```

Comment:
could we have one test to cover scale up/down while doing rolling update with maxSurge and another one to perform two rolling update at the same time

Reply:
Added at integration test.